components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -46,54 +43,29 @@ export const WorkspaceEntry: FunctionComponent<Props> = ({ info, shortVersion })
     );
 
     return (
-        <Item className="whitespace-nowrap py-6 px-6" solid={menuActive}>
+        <Item className="whitespace-nowrap py-2 px-2" solid={menuActive}>
             <ItemFieldIcon>
                 <WorkspaceStatusIndicator instance={info?.latestInstance} />
             </ItemFieldIcon>
-            <ItemField className="w-3/12 flex flex-col my-auto">
+            <ItemField className="w-3/12 my-auto">
                 <a href={startUrl}>
-                    <div className="font-medium text-gray-800 dark:text-gray-200 truncate hover:text-blue-600 dark:hover:text-blue-400">
-                        {workspace.id}
+                    <span className="font-medium text-gray-800 dark:text-gray-200 truncate hover:text-blue-600 dark:hover:text-blue-400 max-w-xs overflow-hidden whitespace-nowrap truncate">
+                        {true ? currentBranch : "Custom description (branch)"}
```

Comment:
**question**: The idea here is when the user opts for renaming a workspace, replace the current branch information with a custom text and append the brand in parentheses after the custom description. Hints or commits are appreciated. 👋 

```
// Before renaming the workspace description
branch-name · 2 minutes ago

// After renaming the workspace description
Custom workspace description (branch-name) · 2 minutes ago

Reply:
I think this would be worth experimenting with, to be honest. It would also be interesting to consider how you might name a workspace when you start it, which could default to a name or description like we do today based on the Git context like PR or a specific branch.